sklearn/cluster/tests/test_optics.py

```diff
@@ -99,15 +101,26 @@ def test_extract_xi():
     X = np.vstack((C1, C2, C3, C4, C5, np.array([[100, 100]]), C6))
     expected_labels = np.r_[[2] * 5, [0] * 5, [1] * 5, [3] * 5, [1] * 5, -1, [4] * 5]
     X, expected_labels = shuffle(X, expected_labels, random_state=rng)
+    X = sparse.lil_matrix(X) if metric == "euclidean" else X
```

Comment:
I don't understand that. Is it to test on dense and sparse ? or sparse doesn't work with minkowski ?

Reply:
Yes this was done due to sparse matrices not being supported by scipy distance metrics. I've included a clarifying line in the docstring description of `metric` to highlight this.